client_test/stub_test.py

```diff
@@ -240,3 +241,22 @@ def test_registered_web_endpoints(client, servicer):
     stub.webhook(square_webhook)
 
     assert stub.registered_web_endpoints == ["square_webhook"]
+
+
+def test_init_types():
+    with pytest.raises(typeguard.TypeCheckError):
+        Stub(secrets=modal.Secret())  # singular secret to plural argument
+    with pytest.raises(typeguard.TypeCheckError):
+        Stub(secrets=[{"foo": "bar"}])  # not a Secret Object
+    with pytest.raises(typeguard.TypeCheckError):
+        Stub(some_arg=5)  # blueprint needs to use Providers
+    with pytest.raises(typeguard.TypeCheckError):
+        Stub(image=modal.Secret())  # should be an Image
+
+    Stub(
+        image=modal.Image.debian_slim().pip_install("typeguard"),
```

Comment:
why are you installing typeguard?

Reply:
Sorry for the confusion - it's just a random package to make a new image instance that resembles what a user might do, just to have some "real world" value that should *not* cause a type error

Reply:
Could be "pandas" or whatever